warehouse/ingest-core/src/main/java/datawave/ingest/metadata/MetadataWithEarliestDate.java

```diff
@@ -0,0 +1,92 @@
+package datawave.ingest.metadata;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.hadoop.io.Text;
+
+import com.google.common.base.Objects;
+
```

Comment:
Please add a class-level comment describing the purpose of the class, as well as some more method-level documentation throughout the class.